From 6fd326c5aaea1dda1c13119dfe87a4e83da71c5e Mon Sep 17 00:00:00 2001
From: Alexander Filonenko <alexf@motorola.com>
Date: Tue, 27 Oct 2015 15:43:36 -0500
Subject: [PATCH 429/959] IKSWM-9389 input: synaptics_dsx: fix F55 register
 access

Fix data structure utilized for Synaptics F55 register access.

Change-Id: Ieaae4881fd4bc96ef91f2c7a3056f67a7374ec32
Signed-off-by: Alexander Filonenko <alexf@motorola.com>
Reviewed-on: http://gerrit.mot.com/794893
SME-Granted: SME Approvals Granted
SLTApproved: Slta Waiver <sltawvr@motorola.com>
Tested-by: Jira Key <jirakey@motorola.com>
Reviewed-by: Konstantin Makariev <kmakariev@motorola.com>
Reviewed-by: Igor Kovalenko <igork@motorola.com>
Submit-Approved: Jira Key <jirakey@motorola.com>
Reviewed-on: http://gerrit.mot.com/795549
---
 drivers/input/touchscreen/synaptics_dsx_test_reporting.c | 12 ++++++++----
 1 file changed, 8 insertions(+), 4 deletions(-)

diff --git a/drivers/input/touchscreen/synaptics_dsx_test_reporting.c b/drivers/input/touchscreen/synaptics_dsx_test_reporting.c
index e7716d36b0e..52e34afb139 100644
--- a/drivers/input/touchscreen/synaptics_dsx_test_reporting.c
+++ b/drivers/input/touchscreen/synaptics_dsx_test_reporting.c
@@ -1657,8 +1657,10 @@ struct f55_control_0 {
 			unsigned char f55_c0_guard_disable:1;
 			unsigned char f55_c0_b5_b7:3;
 		} __packed;
-		unsigned char data[1];
-		unsigned short address;
+		struct {
+			unsigned char data[1];
+			unsigned short address;
+		} __packed;
 	};
 };
 
@@ -1667,8 +1669,10 @@ struct f55_control_8 {
 		struct {
 			unsigned char f55_c8_pattern_type;
 		} __packed;
-		unsigned char data[1];
-		unsigned short address;
+		struct {
+			unsigned char data[1];
+			unsigned short address;
+		} __packed;
 	};
 };
 
-- 
2.11.0

